### ISSUE TITLE
BatteryPool should log exact reasons when components are marked as `NOT_WORKING`

### ISSUE THREAD

**Issue by shsms** [frequenz-floss/frequenz-sdk-python#546]
When BatteryPoolStatus/BatteryStatusTracker mark a battery/inverter pair as `NOT_WORKING`, it needs to log the exact reasons - what check failed, what the failure value was, etc.

Having this would be very helpful when debugging production/deployment issues.

**Comment by Marenz** [comment]
I am not sure what is to do for this ticket. [Every single method ](https://github.com/frequenz-floss/frequenz-sdk-python/blob/v0.x.x/src/frequenz/sdk/actor/power_distributing/_battery_status.py#L235)that indirectly causes a `NOT_WORKING` status, also outputs the reason why it does that. 
What else do we want?

**Comment by tiyash-basu-frequenz** [comment]
> I am not sure what is to do for this ticket. [Every single method ](https://github.com/frequenz-floss/frequenz-sdk-python/blob/v0.x.x/src/frequenz/sdk/actor/power_distributing/_battery_status.py#L235)that indirectly causes a `NOT_WORKING` status, also outputs the reason why it does that.

True. Every method logs the reason behind failure. Including [the one for inverters](https://github.com/frequenz-floss/frequenz-sdk-python/blob/v0.x.x/src/frequenz/sdk/actor/power_distributing/_battery_status.py#L244).

**Comment by llucax** [comment]
Then maybe it was a NOP. @shsms any final words before we close this?
